src/System.Private.CoreLib/src/System/AppContext/AppContext.cs

```diff
@@ -7,29 +7,20 @@
 using System.IO;
 using System.Reflection;
 using System.Runtime.Versioning;
+using System.Threading;
 
 namespace System
 {
     public static class AppContext
     {
-        [Flags]
-        private enum SwitchValueState
-        {
-            HasFalseValue = 0x1,
-            HasTrueValue = 0x2,
-            HasLookedForOverride = 0x4,
-            UnknownValue = 0x8 // Has no default and could not find an override
-        }
-        private static readonly Dictionary<string, SwitchValueState> s_switchMap = new Dictionary<string, SwitchValueState>();
+        private static Dictionary<string, object> s_localStore = new Dictionary<string, object>();
```

Comment:
I think we should use a better name here. "local store" is very generic. I know it was called like this in AppDomain. In a way I also don't like the GetData/SetData as that is also very generic name, but that's public API, so not much we can do...
Hmm...
Maybe at least for consistency with the public API let's call this "data store"... still not great... but a little better at least.